src/serializer.ts

```diff
@@ -17,14 +19,156 @@ interface PaymentJSON {
  * Provides functionality to serialize from protocol buffers to JSON objects.
  */
 class Serializer {
+/**
+   * Convert a Transaction to a JSON representation.
+   *
+   * @param {proto.Transaction} transaction A Transaction to convert.
+   * @returns {Object} The Transaction as JSON.
+   */
+  public static transactionToJSON(
+    transaction: Transaction
+  ): object | undefined {
```

Comment:
Do you mean instead of `null`? I don't really have an opinion.

I tend to think of `undefined` as, the inputs were invalid and the output was not defined. 
Whereas null is: the inputs were valid but the output is nothing. 

I don't know if that's a great mental model. Happy to change if you and other reviewers feel strongly.

Reply:
My mental model for `undefined` and `null` in JS is:
- `undefined` means the JS engine did not assign a value to this particular variable
- `null` means there was a programmer that explicitly made a decision to clear out that variable.

I think it would be poor form to explicitly set a variable to `undefined`, but we have a function here.

For early returning from a function, `undefined` seems right. `void` is the type in Typescript that is commonly used for a function that returns nothing, so maybe that is more appropriate.

Reply:
> My mental model for undefined and null in JS is:
> - undefined means the JS engine did not assign a value to this particular variable
> - null means there was a programmer that explicitly made a decision to clear out that variable.

This matches mine, I think. IIUC, you're in favor of leaving this as is then?

Reply:
Yeah, I'm ok with leaving as is. I do think `return` is more idiomatic than `return undefined`, so thanks for that change.